src/hooks/use-media-query.ts

```diff
@@ -0,0 +1,20 @@
+import { useState, useMemo, useEffect } from 'react'
+
+export function useMediaQuery(query: string): boolean {
```

Comment:
This change would result in a media query listener for every usage of the hook, which is probably fine in this case since there is only one usage.

Reply:
Readability is more important. Even if it does cause problems in the future, caching `MediaQueryList` with `query` as the key is very easy.

Reply:
I think you might be misunderstanding what I'm saying—the issue I'm raising is not with the hook itself but rather the local state (1 listener for every hook usage) vs. context provider state (you could use it 10 times and if the state is managed in a context provider, it'll be 1 listener), i.e. if you call `useDarkMode()` 10 times then that's 9 unnecessary listeners with this approach, but given you're going for a minimalist approach, I imagine there won't be much feature development in that direction, which kind of relates to my choice to use a union type instead of a boolean: using a union type `'dark' | 'light'` could add support for a third option `'system'` in the event where you'd like to support a user choosing an explicit theme instead of using the operating system default, kinda like how the MDN website does it, though again, considering you're going for a minimalist approach with this extension, it may not make sense for your app.

<img width="1177" alt="image" src="https://github.com/user-attachments/assets/ea75d511-2965-45d9-b748-c7c7b248d513">

I'm fine with your approach, just thought I'd mention it in case you need to use the hook more than once, as you'll be creating duplicate listeners with your changes.

Reply:
IMHO, all of your concerns are ridiculous for modern computers and won't be a problem in 99% of cases. No offense, but honestly, only newbies care about this and I'm not going to argue about it any further.

Reply:
I'm just surfacing an inefficiency with your approach and clarifying your misunderstanding of my previous comment and contribution, you can take it or leave it. It looks like only one of us thinks this is an argument. I can tell that an alternative opinion to yours has gotten you so emotionally triggered that you felt the need to result to personal attacks. 

And trust me, I already know after being in this codebase that a few extra event listeners are definitely not going to be the bottleneck of this project—the amount of time it takes such a small project to build is kind of unreasonable and very _extra-_, if you will.

Forgive me for highlighting a concern with your approach and pulling you away from the very important "minimalist" task of rewriting all of lodash and DefinitelyTyped and shipping it as 100+ separate libraries and adding them as dependencies to this project. Carry on.

Reply:
If you care about this level of inefficiency, then you haven't learned to program yet, that's all I know. If you think I'm wrong, that's fine, you can ask someone else, maybe they can teach you. I don't want this low-level argument to continue wasting everyone's time.